### ISSUE TITLE
Bot listening - can it listen continously?

### ISSUE THREAD

**Issue by Mukozuke** [Adri6336/gpt-voice-conversation-chatbot#18]
Hi again :)

So I wonder if it's feasible to make it so the bot is continuously listening to the mic, instead of triggering the recording for every prompt?
The conversation would feel much more natural and fluid if it didn't require pressing any buttons.
Perhaps you are familiar with Neuro-sama AI VTuber which seems to listen to conversation partners all the time but since its creator made it its job, he doesn't want to share details how it works exactly. 

Perhaps to start with we could have it so after the TTS of choice stops talking back to us, automatically triggers recording again?
I think it would require either removing the time window during which you need to say your prompt as currently recording stops after short while if you don't speak or at least prolonging that window significantly.

Just some food for thoughts. I really like the project and would love to see it grow and get even better than it already is!

**Comment by cyberoon** [comment]
I thank the author for such an important project.
I would also be glad if the author added the function of constant wiretapping of the microphone. Or, when contacted by name, he switched to listening mode. :)

**Comment by Adri6336** [comment]
Heyo!

Ikr, that feature was one of the first that I wanted to add. I opted to go away from it because I couldn't find an on-computer solution that would be good enough for the transcription (lots of mistakes with the one I used). I decided against adding it because the only option available to me at the time was to send all recordings to Google, which means that a legit stream of continuous wiretapping would be sent to a third party. Irrespective to how Google would respond to being constantly requested (it may make them rate-limit or block an IP address), it had serious privacy concerns for the user and anyone around them. Thus, I decided to go with a manual keyboard switch for the time being. 

In the future if I can find a good solution that recognizes a wake-word without using an online service, I would absolutely love to add it! It would be like an Echo dot that you actually want to interact with lol
